### PR TITLE
Updated to use hpp-skel

### PR DESCRIPTION
**Pull request by flippmoke**
Updated to use the layout of https://github.com/mapbox/hpp-skel

- Updates unit tests to use catch.hpp
- Added framework for benchmarks (will be added soon after)
- Updated to use cmake
- Added silencing of `-Wfloat-equal` errors in `point.hpp`
- Added codecov
- Updated travis build scripts

/cc @springmeyer @artemp @jfirebaugh @GretaCB @mapsam

### CODE REVIEW

**Comment by flippmoke** [comment]
I should note that I added the `.clang-format` file, but didn't run `make format`. I can run this as well in this PR, but wanted to visually make it easier for others to see differences made else where prior to this.

File: include/mapbox/geometry/point.hpp [hunk 1]
```diff
@@ -19,12 +19,17 @@ struct point
     T y;
 };
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
```
Comment (GretaCB) [inline include/mapbox/geometry/point.hpp]: Curious why ignored "-Wfloat-equal"?

File: scripts/liftoff.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
```
Comment (GretaCB) [inline scripts/liftoff.sh]: You probs dont need this file at all

File: include/mapbox/geometry/point.hpp (hunk 1 above)
Comment (flippmoke) [reply include/mapbox/geometry/point.hpp]: If we do not, then compare operations on any base class using a `double` or `float` type would have a compile warning and fail. This is a known issue that is being considered in #49. Until we figure out a great way to resolve this, I don't want to cause compilation failures.

**Pull request merged**